src/main/java/com/hazelcast/web/HazelcastInstanceLoader.java

```diff
@@ -70,26 +69,57 @@ public static HazelcastInstance createInstance(final ClusteredSessionService ses
         final String configLocation = properties.getProperty(CONFIG_LOCATION);
         final String useClientProp = properties.getProperty(USE_CLIENT);
         final String clientConfigLocation = properties.getProperty(CLIENT_CONFIG_LOCATION);
-        final boolean useClient = !isEmpty(useClientProp) && Boolean.parseBoolean(useClientProp);
-        URL configUrl = null;
-        if (useClient && !isEmpty(clientConfigLocation)) {
-            configUrl = getConfigURL(sessionService.getFilterConfig(), clientConfigLocation);
-        } else if (!isEmpty(configLocation)) {
-            configUrl = getConfigURL(sessionService.getFilterConfig(), configLocation);
-        }
-        String sessionTTLConfig = properties.getProperty(SESSION_TTL_CONFIG);
+        final boolean useClient = !isNullOrEmptyAfterTrim(useClientProp) && Boolean.parseBoolean(useClientProp);
+        final String mapName = properties.getProperty(MAP_NAME);
+        final String sessionTTL = properties.getProperty(SESSION_TTL_CONFIG);
+
+        URL configUrl = getConfigUrl(sessionService, configLocation, clientConfigLocation, useClient);
```

Comment:
I would normally get rid of `getConfigUrl` and put the finding config file into its own place.... see my comment below.